woocommerce/woocommerce-google-analytics-integration
# Upgrade to use Node.js 20 and bump npm dependencies

### Changes proposed in this Pull Request:

The end-of-life for Node.js v18 is April 30, 2025, which is still about 10 months away, but there should be no harm in aligning with the same version used in WooCommerce core early!

This PR:

- Upgrade to use Node.js v20
- Bump dependencies
   - Add `@wordpress/url` to dev dependencies as it's no longer a dependency under `@wordpress/scripts`

### Checks:


### Detailed test instructions:

1. Set Node.js to v20
2. Run `npm install` to see if it can finish without errors
3. Check if the following scripts work well
   - `npm start`
   - `npm run dev`
   - `npm run build`
   - `npm run lint:js`
4. Inspect [the E2E test result on GitHub Actions](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/9709927385/job/26799575793) or run it locally to see if all tests can pass

### Changelog entry

- Dev - Upgrade to Node.js 20.
- Dev - Upgrade npm dependencies.
